cmd/utils/nodecmd/dumpconfigcmd.go

```diff
@@ -120,22 +146,652 @@ func makeConfigNode(ctx *cli.Context) (*node.Node, klayConfig) {
+			log.Fatalf("Option %q: %v", utils.NodeKeyHexFlag.Name, err)
+		}
+		cfg.PrivateKey = key
+	}
+}
+
+// setNAT creates a port mapper from command line flags.
+func setNAT(ctx *cli.Context, cfg *p2p.Config) {
+	if ctx.GlobalIsSet(utils.NATFlag.Name) {
+		natif, err := nat.Parse(ctx.GlobalString(utils.NATFlag.Name))
+		if err != nil {
+			log.Fatalf("Option %s: %v", utils.NATFlag.Name, err)
+		}
+		cfg.NAT = natif
+	}
+}
+
+// setListenAddress creates a TCP listening address string from set command
+// line flags.
+func setListenAddress(ctx *cli.Context, cfg *p2p.Config) {
+	if ctx.GlobalIsSet(utils.ListenPortFlag.Name) {
+		cfg.ListenAddr = fmt.Sprintf(":%d", ctx.GlobalInt(utils.ListenPortFlag.Name))
+	}
+
+	if ctx.GlobalBool(utils.MultiChannelUseFlag.Name) {
+		cfg.EnableMultiChannelServer = true
+		SubListenAddr := fmt.Sprintf(":%d", ctx.GlobalInt(utils.SubListenPortFlag.Name))
+		cfg.SubListenAddr = []string{SubListenAddr}
+	}
+}
+
+func convertNodeType(nodetype string) common.ConnType {
+	switch strings.ToLower(nodetype) {
+	case "cn", "scn":
+		return common.CONSENSUSNODE
+	case "pn", "spn":
+		return common.PROXYNODE
+	case "en", "sen":
+		return common.ENDPOINTNODE
+	default:
+		return common.UNKNOWNNODE
+	}
+}
+
+// setBootstrapNodes creates a list of bootstrap nodes from the command line
+// flags, reverting to pre-configured ones if none have been specified.
+func setBootstrapNodes(ctx *cli.Context, cfg *p2p.Config) {
+	var urls []string
+	switch {
+	case ctx.GlobalIsSet(utils.BootnodesFlag.Name):
+		logger.Info("Customized bootnodes are set")
+		urls = strings.Split(ctx.GlobalString(utils.BootnodesFlag.Name), ",")
+	case ctx.GlobalIsSet(utils.CypressFlag.Name):
+		logger.Info("Cypress bootnodes are set")
+		urls = params.MainnetBootnodes[cfg.ConnectionType].Addrs
+	case ctx.GlobalIsSet(utils.BaobabFlag.Name):
+		logger.Info("Baobab bootnodes are set")
+		// set pre-configured bootnodes when 'baobab' option was enabled
+		urls = params.BaobabBootnodes[cfg.ConnectionType].Addrs
+	case cfg.BootstrapNodes != nil:
+		return // already set, don't apply defaults.
+	case !ctx.GlobalIsSet(utils.NetworkIdFlag.Name):
+		if utils.NodeTypeFlag.Value != "scn" && utils.NodeTypeFlag.Value != "spn" && utils.NodeTypeFlag.Value != "sen" {
+			logger.Info("Cypress bootnodes are set")
+			urls = params.MainnetBootnodes[cfg.ConnectionType].Addrs
+		}
+	}
+
+	cfg.BootstrapNodes = make([]*discover.Node, 0, len(urls))
+	for _, url := range urls {
+		node, err := discover.ParseNode(url)
+		if err != nil {
+			logger.Error("Bootstrap URL invalid", "kni", url, "err", err)
+			continue
+		}
+		if node.NType == discover.NodeTypeUnknown {
+			logger.Debug("setBootstrapNode: set nodetype as bn from unknown", "nodeid", node.ID)
+			node.NType = discover.NodeTypeBN
+		}
+		logger.Info("Bootnode - Add Seed", "Node", node)
+		cfg.BootstrapNodes = append(cfg.BootstrapNodes, node)
+	}
+}
+
+// SetNodeConfig applies node-related command line flags to the config.
```

Comment:
```suggestion
// setNodeConfig applies node-related command line flags to the config.
```
Seems like a minor typo